dvc/remote/gdrive.py

```diff
@@ -324,10 +325,40 @@ def _create_remote_dir(self, parent_id, title):
 
     @gdrive_retry
     def _delete_remote_file(self, remote_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": remote_id}
         # it does not create a file on the remote
         item = self.drive.CreateFile(param)
-        item.Delete()
+
+        try:
+            item.Trash() if self._trash_only else item.Delete()
+        except ApiRequestError as exc:
+            if (
+                not exc.error.get("code", None)
+                and self.list_params["corpora"] != "drive"
+            ):
+                raise
+            if (
+                exc.error.get("code", None) == 403
```

Comment:
I wonder if there is a case with it returning None?

Reply:
It seems it can't have 'None'. Finally, exactly this check is remove and upper condition is simplified to:
```
            http_error_code = exc.error.get("code", 0)
            if (
                http_error_code != 403
                or self.list_params["corpora"] != "drive"
            ):
                raise
```